src/app/technical-record/technical-record.component.ts

```diff
@@ -1,116 +1,116 @@
-  }
-
-  togglePanel() {
-    for (const panel of this.panels) {
-      panel.isOpened = !this.allOpened;
-    }
-    this.allOpened = !this.allOpened;
-  }
-
-  hasSecondaryVrms(vrms) {
-    return vrms && vrms.length > 1 && vrms.filter((vrm) => vrm.isPrimary === false).length > 0;
-  }
-
-  editTechRecord() {
-    this.changeViewState.emit(VIEW_STATE.EDIT);
-  }
-
-  cancelTechRecordEdit() {
-    this.changeViewState.emit(VIEW_STATE.VIEW_ONLY);
-    this.activeRecord = JSON.parse(JSON.stringify(this.activeRecord));
-    this.adrDisplayParams = { ...{ showAdrDetails: !!this.activeRecord.adrDetails } };
-
-    const controlNames = Object.keys(this.techRecord.controls);
-    for (let index = 0; index < controlNames.length; index++) {
-      this.techRecord.removeControl(controlNames[index]);
-    }
-    this.techRecord.updateValueAndValidity();
-  }
-
-  onSaveChanges({ valid, value }: { valid: boolean; value: any }) {
-    // if (valid) { TODO: Re-enable during ADR Validation ticket
-
-    const editedRecord: TechRecord = this.allowedValues.mapControlValuesToDataValues(
-      JSON.parse(JSON.stringify(value))
-    );
-    const mergedRecord: TechRecord = { ...this.activeRecord, ...editedRecord };
-
-    const dialogRef = this.dialog.open(AdrReasonModalComponent, {
-      width: '600px',
-      data: {
-        context: 'Enter reason for changing technical record'
-      }
-    });
-
-    dialogRef.afterClosed().subscribe((result) => {
-      if (result && result.isSave) {
-        mergedRecord.reasonForCreation = result.data;
-        this.submitTechRecord.emit(mergedRecord);
-      }
-    });
-    // }
-  }
-}
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output
+} from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { MatDialog } from '@angular/material/dialog';
+
+import { AdrReasonModalComponent } from '@app/shared/adr-reason-modal/adr-reason-modal.component';
+import { TechRecordHelpersService } from '@app/technical-record/tech-record-helpers.service';
+
+import { TechnicalRecordValuesMapper } from './technical-record.mapper';
+import { TechRecord } from './../models/tech-record.model';
+import { MetaData } from '@app/models/meta-data';
+import { VehicleTechRecordModel } from '@app/models/vehicle-tech-record.model';
+import { TestResultModel } from '@app/models/test-result.model';
+import { VIEW_STATE } from '@app/app.enums';
+
+@Component({
+  selector: 'vtm-technical-record',
+  templateUrl: './technical-record.component.html',
+  styleUrls: ['../app.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class TechnicalRecordComponent implements OnInit {
+  showAdrDetails: boolean;
+  adrDisplayParams: { [key: string]: boolean };
+  techRecord: FormGroup;
+
+  allOpened = false;
+  panels: { panel: string; isOpened: boolean }[] = [
+    { panel: 'panel1', isOpened: false },
```

Comment:
need to have more meaningful name for panels maybe

Reply:
@JonLazariniDD we will do this as part of a tech debt to refactor the existing accordion. this file should not be part of 10278 commit but for the code change in line 51

Reply:
As agreed, subsequent refactor if needed as code was done prior to ADR. For reference: https://jira.dvsacloud.uk/browse/CVSB-15024 covered in 10163